tests/test_huawei.py

```
@@ -0,0 +1,392 @@
+    assert asyncio.run(_acall(api, "login")) is True
+    assert api._session_cookie == "Cookie=sid=deadbeef:Language:portuguese:id=1"  # noqa: SLF001
+    assert any("GetRandCount" in c for c in calls)
+    assert any("login.cgi" in c for c in calls)
+
+
+def test_login_failure_raises(base_url: URL, monkeypatch: pytest.MonkeyPatch) -> None:
+    """Missing sid cookie raises CannotAuthenticate."""
+
+    class _Resp:
+        status = 200
+        headers: dict[str, str] = {}
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)